go/ct/spc/specification.go

```diff
@@ -2045,15 +2051,17 @@ func callEffect(s *st.State, addrAccessCost vm.Gas, op OpCode) {
 		memoryExpansionCost = outputMemoryExpansionCost
 	}
 
+	isValueZero := value.IsZero()
```

Comment:
Delegate calls are never transferring any value since the source and destination account are the same. So I think it would be simpler to not set the value in line 2037 to something in case of a delegate call, and instead leave it to be 0.

In the condition starting at line 2119 you can then update the value to the value taken from the call context.